### PR TITLE
 Remove code to extract LZMA on first run

### PR DESCRIPTION
**Pull request by dsplaisted**
Remove LZMA extraction code.

This is the first part of #9910.  We will also need to remove the actual LZMA from core-sdk after this PR is merged (and integrated to core-sdk).

### CODE REVIEW

**Comment by dsplaisted** [comment]
@dotnet-bot test Linux_NoSuffix arm Release Build

**Comment by dsplaisted** [comment]
@dotnet-bot test Linux_NoSuffix arm Release Build

File: test/dotnet-msbuild.Tests/GivenDotnetMSBuildBuildsProjects.cs [hunk 1]
```diff
@@ -192,13 +192,13 @@ private string[] GetArgsForMSBuild(Func<bool> sentinelExists, out Telemetry.Tele
         }
     }
 
-    public sealed class MockNuGetCacheSentinel : IFirstTimeUseNoticeSentinel
+    public sealed class MockFirstTimeUseCacheSentinel : IFirstTimeUseNoticeSentinel
```
Comment (wli3) [inline test/dotnet-msbuild.Tests/GivenDotnetMSBuildBuildsProjects.cs]: nit, may be FirstTimeUseNoticeSentinel. "First Time Use Cache" is not what happens now

**Comment by wli3** [comment]
the pr is good but have you tested end to end once especially on mac or linux? it should be ok but will it get in forever permission denied? usually whenever people touch first run, you get a bug like that 

also we add in nuget fallback folder for many build, will they still be fine at the current state?

**Comment by wli3** [comment]
also, FYI, the current fallback folder is at a sudo location, if the targeting packs can be added afterwards, we need to remember adding option storing them in a user location, which is in the logic removing.

**Comment by dsplaisted** [comment]
@dotnet-bot test this please

**Comment by dsplaisted** [comment]
@dotnet-bot test Debian8.2 x64 Debug Build

**Pull request merged**